clients/src/main/java/org/apache/kafka/common/requests/AbstractResponse.java

```diff
@@ -31,7 +31,7 @@
     public static final int DEFAULT_THROTTLE_TIME = 0;
 
     protected Send toSend(String destination, ResponseHeader header, short apiVersion) {
-        return new NetworkSend(destination, serialize(apiVersion, header));
+        return new StructSend(destination, header.toStruct(), toStruct(apiVersion));
```

Comment:
do we need to cache serialized response? If the saving is not significant, can we just keep the upstream code?

Reply:
I wanted to use Buffer pools to reuse the underlying buffers once the send is in #completed() state. But that change can go in a different PR.